### PR TITLE
add actions to picture-glance entities

### PR DESCRIPTION
**Pull request by iantrich**
Docs: https://github.com/home-assistant/home-assistant.io/pull/10498

### CODE REVIEW

**Comment by iantrich** [comment]
will add docs once https://github.com/home-assistant/home-assistant-polymer/pull/3791 and its corresponding docs are merged

File: src/panels/lovelace/cards/hui-picture-glance-card.ts [hunk 1]
```diff
@@ -189,8 +192,10 @@ class HuiPictureGlanceCard extends LitElement implements LovelaceCard {
 
     return html`
       <ha-icon
-        .entity="${stateObj.entity_id}"
-        @click="${dialog ? this._openDialog : this._callService}"
+        @ha-click=${this._handleTap}
+        @ha-hold=${this._handleHold}
+        .longPress=${longPress()}
+        .config=${entityConf}
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-picture-glance-card.ts]: No need for this, can use `this._config` in the event handler.

File: src/panels/lovelace/cards/hui-picture-glance-card.ts [hunk 2]
```diff
@@ -129,16 +126,17 @@ class HuiPictureGlanceCard extends LitElement implements LovelaceCard {
...
               this._config.tap_action ||
                 this._config.hold_action ||
                 this._config.camera_image
             ),
-          })}"
+          })}
           @ha-click=${this._handleTap}
           @ha-hold=${this._handleHold}
           .longPress=${longPress()}
+          .config=${this._config}
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-picture-glance-card.ts]: No need for this, can use `this._config` in the event handler.

File: src/panels/lovelace/cards/hui-picture-glance-card.ts [hunk 3]
```diff
@@ -206,20 +211,14 @@ class HuiPictureGlanceCard extends LitElement implements LovelaceCard {
...
-
-  private _handleHold() {
-    handleClick(this, this.hass!, this._config!, true);
-  }
-
-  private _openDialog(ev: MouseEvent): void {
-    fireEvent(this, "hass-more-info", { entityId: (ev.target as any).entity });
+  private _handleTap(ev: MouseEvent): void {
+    const config = (ev.currentTarget as any).config as any;
+    handleClick(this, this.hass!, config, false);
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-picture-glance-card.ts]:
```suggestion
    handleClick(this, this.hass!, this._config, false);
```

File: src/panels/lovelace/cards/hui-picture-glance-card.ts [hunk 4]
```diff
@@ -206,20 +211,14 @@ class HuiPictureGlanceCard extends LitElement implements LovelaceCard {
...
   }
 
-  private _callService(ev: MouseEvent): void {
-    toggleEntity(this.hass!, (ev.target as any).entity);
+  private _handleHold(ev: MouseEvent): void {
+    const config = (ev.currentTarget as any).config as any;
+    handleClick(this, this.hass!, config, true);
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-picture-glance-card.ts]:
```suggestion
    handleClick(this, this.hass!, this._config, false);
```

File: src/panels/lovelace/cards/hui-picture-glance-card.ts (hunk 1 above)
Comment (iantrich) [reply src/panels/lovelace/cards/hui-picture-glance-card.ts]: This is the config for just this entity. We don't actions for the parent

**Comment by iantrich** [comment]
@bramkragten passing the config is so that you can do it per entity with the same function

File: src/panels/lovelace/cards/hui-picture-glance-card.ts (hunk 1 above)
Comment (bramkragten) [reply src/panels/lovelace/cards/hui-picture-glance-card.ts]: Ah I see :-)

**Comment by bramkragten** [comment]
> @bramkragten passing the config is so that you can do it per entity with the same function

Yeah see it now, was to quick 👍

**Comment by iantrich** [comment]
@bramkragten let me know if you need anything else on this. Was hoping to get it into the beta

**Pull request merged**